app/models/concerns/direct_db_conn.rb

```diff
@@ -0,0 +1,65 @@
+module DirectDBConn
+  class << self
+    def insert_on_db(objects, table)
+      string_params = generate_string_params_template(objects.length)
+      column_names = objects.first.attributes.keys.join(', ')
+      object_values = objects.map{|o| "(#{convert_values_to_insert_syntax(o)})"}
+      # sql_query_array = ["INSERT INTO #{table} (#{column_names}) VALUES #{string_params}", object_values].flatten
+      # TODO sanitize
+      sql_query_array = ["INSERT INTO #{table} (#{column_names}) VALUES #{object_values.join(', ')}"]
+
+      # Rescue because this uniqueness validation is already inserted on the model
+      execute_sql_query_array(sql_query_array) rescue ActiveRecord::RecordNotUnique
+    end
+
+    def check_uniqueness_on_db(objects, table, field, row_start, columns=[])
```

Comment:
See my later comment on how this could be refactored. Seems like you should only need to call this method once for phone and phone2.